go/apps/api/routes/v2_apis_get_api/handler.go

```diff
@@ -65,34 +68,37 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
 		return err
 	}
 
-	// Get API from database
-	api, err := db.Query.FindApiByID(ctx, h.DB.RO(), req.ApiId)
+	api, hit, err := h.Caches.LiveApiByID.SWR(ctx, cache.ScopedKey{WorkspaceID: auth.AuthorizedWorkspaceID, Key: req.ApiId}, func(ctx context.Context) (db.FindLiveApiByIDRow, error) {
+		return db.Query.FindLiveApiByID(ctx, h.DB.RO(), req.ApiId)
+	}, caches.DefaultFindFirstOp)
+
 	if err != nil {
 		if db.IsNotFound(err) {
 			return fault.New("api not found",
 				fault.Code(codes.Data.Api.NotFound.URN()),
 				fault.Internal("api not found"), fault.Public("The requested API does not exist or has been deleted."),
 			)
 		}
+
 		return fault.Wrap(err,
 			fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
 			fault.Internal("database error"), fault.Public("Failed to retrieve API information."),
 		)
 	}
 
-	// Check if API belongs to the authorized workspace
-	if api.WorkspaceID != auth.AuthorizedWorkspaceID {
-		return fault.New("wrong workspace",
+	if hit == cache.Null {
+		return fault.New("api not found",
 			fault.Code(codes.Data.Api.NotFound.URN()),
-			fault.Internal("wrong workspace, masking as 404"), fault.Public("The requested API does not exist or has been deleted."),
+			fault.Internal("api not found"),
+			fault.Public("The requested API does not exist or has been deleted."),
 		)
 	}
 
-	// Check if API is deleted
-	if api.DeletedAtM.Valid {
-		return fault.New("api not found",
+	// Check if API belongs to the authorized workspace
+	if api.WorkspaceID != auth.AuthorizedWorkspaceID {
+		return fault.New("wrong workspace",
```

Comment:
this is now leaking the existence of the workspace to the user, let's use the same response as for the not-found case above